addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/handler/YeelightHandlerBase.java

```diff
@@ -0,0 +1,231 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.yeelight.handler;
+
+import com.yeelight.sdk.device.ConnectState;
+import com.yeelight.sdk.device.DeviceBase;
+import com.yeelight.sdk.device.DeviceFactory;
+import com.yeelight.sdk.device.DeviceStatus;
+import com.yeelight.sdk.enums.DeviceAction;
+import com.yeelight.sdk.enums.DeviceMode;
+import com.yeelight.sdk.enums.DeviceType;
+import com.yeelight.sdk.listeners.DeviceConnectionStateListener;
+import com.yeelight.sdk.listeners.DeviceStatusChangeListener;
+import com.yeelight.sdk.services.DeviceManager;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.HSBType;
+import org.eclipse.smarthome.core.library.types.IncreaseDecreaseType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.yeelight.YeelightBindingConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.openhab.binding.yeelight.YeelightBindingConstants.COLOR_TEMPERATURE_MINIMUM;
+import static org.openhab.binding.yeelight.YeelightBindingConstants.COLOR_TEMPERATURE_STEP;
+
+/**
+ * The {@link YeelightHandlerBase} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Coaster Li (lixin@yeelink.net) - Initial contribution
+ */
+public abstract class YeelightHandlerBase extends BaseThingHandler
+        implements DeviceConnectionStateListener, DeviceStatusChangeListener {
+
+
+    protected Logger mLogger = LoggerFactory.getLogger(YeelightHandlerBase.class);
```

Comment:
Please call your logger `logger` as defined in the coding guidelines.

And consider to make it private and give every class its own instance, now the other class' its log lines seem to originate from `YeelightHandlerBase.class`. By giving every class its own logger you also don't need the `TAG` anymor.